Dashboard: Update side nav designs

samwhale opened the pull request:
## Summary

Updates the side nav designs. [Figma designs](https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=467%3A13745)

## Relevant Technical Choices

The icons have an "active" and "inactive" state. I added separate svgs for each. Each icon has a slightly different active state so I added them separately. If we want we could use styled components to do the same thing.

## To-do

- Change text in side nav links. These are connected to karma tests and can be annoying so I'll do those separately.
- Found a bug where re-sizing the dashboard when visiting the editor settings won't collapse the side nav. Will fix this in a separate pr.

## User-facing changes

- remove side nav right border
- add icons (active + inactive states)
- move `support` and `settings` links to the bottom of the side nav
- change `Editor Settings` link to be just `Settings`
- remove font weight `bold` from the nav link active state

## Testing Instructions

- go to dashboard
- click side nav links to make sure the active states look correct
- resize the window to make sure side nav collapsing still works correctly
- should be keyboard accessible and screen reader accessible

|Interacting with links|Collapsing side nav|
|--|--|
|![side-nav-1](https://user-images.githubusercontent.com/22185279/104767662-c5476c00-5729-11eb-8cb4-f5ef4d648c93.gif)|![side-nav-collapse](https://user-images.githubusercontent.com/22185279/104767672-c9738980-5729-11eb-8f51-d9e20860cfa8.gif)|

---

<!-- Please reference the issue(s) this PR addresses. -->

Fixes #5054

samwhale committed:
Add svgs: normal + active versions

samwhale committed:
Side nav design updates

BrittanyIRL commented on assets/src/dashboard/constants/index.js:
```diff
@@ -21,6 +21,12 @@ import { __ } from '@wordpress/i18n';
+import { ReactComponent as FloppyDisk } from '../../../src/edit-story/icons/floppy_disk.svg';
+import { ReactComponent as FloppyDiskActive } from '../../../src/edit-story/icons/floppy_disk_active.svg';
+import { ReactComponent as Compass } from '../../../src/edit-story/icons/compass.svg';
+import { ReactComponent as CompassActive } from '../../../src/edit-story/icons/compass_active.svg';
```
this is my bad for not explaining this well, there's a section in the design system that should have these icons already. `import { ExploreTemplatesActive, ExploreTemplatesNormal ... } from '../../blah/blah/design-system'` - named 'em that way because they are specific to the dashboard.

samwhale replied:
ah that would be way better, thanks!

samwhale committed:
Move icons and use better syntax

samwhale commented on assets/src/dashboard/constants/index.js:
```diff
@@ -64,22 +72,34 @@ export const ROUTE_TITLES = {
   },
+];
+
+export const SECONDARY_PATHS = [
```
These are the links that are stuck to the bottom of the nav. I separated them out into their own constant.

samwhale committed:
Cleanup - no icon in secondary paths

samwhale committed:
Use styled components to make acive icons

samwhale committed:
Uncomment flag

samwhale committed:
Use design system colors

BrittanyIRL commented on assets/src/design-system/icons/dashboard/index.js:
```diff
@@ -18,8 +18,16 @@
+export { ReactComponent as HomeWithHeart } from './home_with_heart.svg';
 export { ReactComponent as SavedTemplatesActive } from './savedTemplatesActive.svg';
 export { ReactComponent as SavedTemplatesNormal } from './savedTemplatesNormal.svg';
 export { ReactComponent as StoriesActive } from './storiesActive.svg';
```
Since you replaced the need for duplicate SVGs here, can you remove anything in icons/dashboard that is not used for the nav? These are only used there so we don't need the dupes.

samwhale committed:
Merge branch 'main' into feature/5054-side-nav-new-designs

samwhale committed:
Remove unused svgs

samwhale commented on assets/src/design-system/icons/dashboard/index.js:
```diff
@@ -18,9 +18,11 @@
  * Internal dependencies
  */
 
-export { ReactComponent as ExploreTemplatesActive } from './exploreTemplatesActive.svg';
-export { ReactComponent as ExploreTemplatesNormal } from './exploreTemplatesNormal.svg';
-export { ReactComponent as SavedTemplatesActive } from './savedTemplatesActive.svg';
-export { ReactComponent as SavedTemplatesNormal } from './savedTemplatesNormal.svg';
-export { ReactComponent as StoriesActive } from './storiesActive.svg';
-export { ReactComponent as StoriesNormal } from './storiesNormal.svg';
```
These icons are not used in the nav or anywhere in our codebase.

samwhale committed:
Fix tests

samwhale committed:
Empty commit to re-trigger ci

samwhale committed:
Update colors

maxyinger commented on assets/src/dashboard/components/pageStructure/index.js:
```diff
@@ -172,34 +188,68 @@ export function LeftRail() {
+                    })}
+                  >
+                    <NavLinkContent active={path.value === state.currentPath}>
+                      {Icon ? <Icon width="22px" /> : <div />}
```
is the div here doing anything if there's no icon?

samwhale replied:
It's using grid css so the empty div fills the space if there's no icon. There's probably a better way to do this though

maxyinger replied:
oh okay. makes sense. that's fine by me. was just curious

BrittanyIRL replied:
Could also handle this by having the link content start at the second column of the grid by using [grid-template-areas](https://css-tricks.com/snippets/css/complete-guide-grid/). That'd require a bit of a restructure.

maxyinger commented on assets/src/dashboard/components/pageStructure/navigationComponents.js:
```diff
@@ -80,30 +80,15 @@ export const NavLink = styled.a`
-        ? theme.DEPRECATED_THEME.colors.gray900
-        : theme.DEPRECATED_THEME.colors.gray600
-    };
+    color: ${active ? theme.colors.gray['90'] : theme.colors.gray['70']};
```
I think for all of these colors, you can just use the number instead of wrapping it in a string:
```suggestion
    color: ${active ? theme.colors.gray[90] : theme.colors.gray[70]};
```

maxyinger reviewed (approved):
lgtm

samwhale committed:
Update assets/src/dashboard/components/pageStructure/navigationComponents.js

Co-authored-by: Max Yinger <35983235+littlemilkstudio@users.noreply.github.com>

BrittanyIRL commented on assets/src/dashboard/components/pageStructure/index.js:
```diff
@@ -33,7 +33,12 @@ import { BEZIER } from '../../../animation';
 import { resolveRoute, useRouteHistory } from '../../app/router';
-import { BUTTON_TYPES, PRIMARY_PATHS, Z_INDEX } from '../../constants';
+import {
+  BUTTON_TYPES,
```
Let's use the design system button types!

BrittanyIRL commented on assets/src/dashboard/components/pageStructure/index.js:
```diff
@@ -79,8 +84,7 @@ export const LeftRailContainer = styled.nav.attrs({
   width: ${DASHBOARD_LEFT_NAV_WIDTH}px;
-  background: ${({ theme }) => theme.DEPRECATED_THEME.colors.white};
-  border-right: ${({ theme }) => theme.DEPRECATED_THEME.borders.gray50};
+  background: ${({ theme }) => theme.colors.standard.white};
```
theme.colors.bg.primary please, just incase we need to have a light/dark toggle someday.

BrittanyIRL reviewed (changes requested):
I'm really excited to get this updated side nav in use! 

Few things, mostly because we're gutting the old styles out of the dashboard as we go. 

1) Please use the theme.colors.fg and bg whenever possible. This will make future theme updates so much easier to manage. 

2) This is a really prime chance to gut any structure that is unnecessary and refactor. So like, if you want to swap to using css grid for the list items, feel free to remove excess wrappers and restructure. 

3) Let's take this opportunity to make sure the sideNav isn't using _any_ components from the dashboard/components - we can replace the "create new story" button with the one from the design system.  And the Header can get updated to use design system components as well. 

4) Let's verify with Sam that she wants "Editor Settings" back in action, I remember there being a conversation somewhere about swapping to just "settings" but this is why documentation is important.

samwhale committed:
Use design system instead of dashboard components
- move web stories logo to design system
- replace components in 'navigationComponents.js` with components from the design system
- don't use 'inProgress' for secondary links since it doesn't exist yet

samwhale committed:
Do not need ternary anymore

samwhale commented on assets/src/dashboard/components/pageStructure/navigationComponents.js:
```diff
@@ -17,13 +17,12 @@
  */
-import Button from '../button';
-import { TypographyPresets } from '../typography';
+import { Button, Headline, Text } from '../../../design-system';
```
this file only uses base components + components from the design system

samwhale committed:
New icons + new brandPress color

- TODO: add active states in component specific styling

samwhale committed:
Implement new feedback from Sam

samwhale committed:
Trigger build

BrittanyIRL commented on assets/src/dashboard/components/pageStructure/navigationComponents.js:
```diff
@@ -43,72 +42,73 @@ export const Header = styled.h1`
+  ({ theme }) => css`
+    margin-bottom: 0;
+    margin-top: 0;
+    border: 1px solid ${theme.colors.border.defaultActive};
```
2px for now until we sort out how we want to handle needing a 2px wide border on focus so that there's no shift. we'll handle the update in the `Button` for continuity.

BrittanyIRL replied:
Let's also add `    &:focus {
      box-shadow: none;
    }` to this button and to the `NavLink` to override the box-shadow that common css is adding through wordpress. 

Eventually we should be able to control this with the global style overrides in the design system but we're in this kinda murky middle ground. It's something we'll have to make sure is fixed before starting in on the editor.

BrittanyIRL commented on assets/src/dashboard/components/pageStructure/navigationComponents.js:
```diff
@@ -43,72 +42,73 @@ export const Header = styled.h1`
-          ? theme.DEPRECATED_THEME.colors.gray900
-          : theme.DEPRECATED_THEME.colors.gray600
-      };
+export const NavButton = styled(Button)`
```
I'm not seeing this in use anymore?

samwhale replied:
Good catch!

BrittanyIRL commented on assets/src/dashboard/components/pageStructure/navigationComponents.js:
```diff
@@ -43,72 +42,73 @@ export const Header = styled.h1`
+`;
+
+const activeLinkCss = css`
+  * {
```
this is a lot of global selector usage ( `*` ) - is there any way to avoid it?

samwhale replied:
yeah definitely, we can just target the children explicitly 👍

BrittanyIRL commented on assets/src/dashboard/components/pageStructure/navigationComponents.js:
```diff
@@ -43,72 +42,73 @@ export const Header = styled.h1`
+    grid-template-areas: 'icon link link link link';
+
+    * {
+      font-weight: ${theme.typography.weight.bold};
```
does giving the NavLink `isBold` as a prop not work? It's using the `Text` component so that should be baked in.

samwhale replied:
Unfortunately no, since the child is a `<Text>` element BUT we can add it to the child 👍

BrittanyIRL commented on assets/src/design-system/images/index.js:
```diff
@@ -0,0 +1,20 @@
+/**
+ * Internal dependencies
+ */
+export { ReactComponent as WebStoriesLogo } from './webStoriesFullLogo.svg';
```
I'm torn about this being here? We've been trying to keep images in assets/images and i can't remember why this was specifically in dashboard/images instead.

samwhale replied:
I can put it back and we can always move it later 👍

BrittanyIRL reviewed (commented):
Great work on this! It looks great and is very thorough. 

Few small things (this is the first major thing that's getting redesigned and it's pointing out some overlooked issues that didn't exist in storybook). 

Could you add `ThemeGlobals` to the dashboard/app/index like so? 
![Screen Shot 2021-01-22 at 2 49 57 PM](https://user-images.githubusercontent.com/10720454/105552416-1d99e300-5cc1-11eb-83e4-415f3c5f15ee.png)

This will allow the outline override for just the keyboard to exist. 

Unfortunately it doesn't take into consideration the box-shadow added by common, so that's why i think for now we should manually override that for this update and handle it separate.

samwhale committed:
Small fixes
- remove blurry outline from wordpress styling
- use `isBold` rather than using css
- 2px border for button
- don't use `*` selector
- move images back to assets/dashboard

samwhale committed:
Transition all the time

zachhale commented on assets/src/dashboard/components/pageStructure/navigationComponents.js:
```diff
@@ -43,72 +42,76 @@ export const Header = styled.h1`
   padding: 0;
 `;
 
 export const NavListItem = styled.li`
```
The spacing between items in this list seem a little more compact than in the Figma designs by 4ish pixels both in the top lists and the Settings/Support list at the bottom.

samwhale replied:
I'll fix that!

zachhale reviewed (commented):
Looks good other than a little spacing inconsistency.

BrittanyIRL commented on readme.txt:
```diff
@@ -99,7 +99,7 @@ Web Stories created using Web Stories for WordPress use what’s provided in the
 
 There are two ways to include analytics in your Web Stories:
 
-* Add your Google Analytics ID in the “Editor Settings”.
```
nice catch!

BrittanyIRL reviewed (approved):
This looks so good, thank you for the attention to detail and the back and forth.  This will go to QA to make sure there's no regressions and then we can merge and go to UAT where Sam and the Huge team can give us more feedback.

samwhale committed:
Minor styling updates
- increase margin in list items to match figma designs
- remove animation for border color

samwhale committed:
re-run ci

samwhale committed:
try again

samwhale committed:
TRY AGAIN

samwhale committed:
PLEASE

swissspidy committed:
Merge branch 'main' into feature/5054-side-nav-new-designs

swissspidy committed:
Fix duplicate import after merge

samwhale committed:
Fix the links!

swissspidy committed:
Merge branch 'main' into feature/5054-side-nav-new-designs
